netbox/extras/management/commands/buildschema.py

```diff
@@ -0,0 +1,57 @@
+from json import dumps as json_dumps
+from json import loads as json_loads
+from jinja2 import FileSystemLoader, Environment
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from dcim.choices import (
+    DeviceAirflowChoices, SubdeviceRoleChoices, ConsolePortTypeChoices, PowerPortTypeChoices,
+    PowerOutletTypeChoices, PowerOutletFeedLegChoices, InterfaceTypeChoices, InterfacePoEModeChoices,
+    InterfacePoETypeChoices, PortTypeChoices, WeightUnitChoices
+)
+
+
+class Command(BaseCommand):
+    help = "Generate the NetBox validation schemas."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--console',
+            action='store_true',
+            help="Print the generated schema to stdout"
+        )
+        parser.add_argument(
+            '--file',
+            action='store_true',
+            help="Print the generated schema to the generated file"
+        )
+
+    def handle(self, *args, **kwargs):
+        schemas = {}
+        schemas["airflow_choices"] = json_dumps(DeviceAirflowChoices.values())
+        schemas["weight_unit_choices"] = json_dumps(WeightUnitChoices.values())
+        schemas["subdevice_role_choices"] = json_dumps(SubdeviceRoleChoices.values())
+        schemas["console_port_type_choices"] = json_dumps(ConsolePortTypeChoices.values())  # console-ports and console-server-ports
+        schemas["power_port_type_choices"] = json_dumps(PowerPortTypeChoices.values())
+        schemas["power_outlet_type_choices"] = json_dumps(PowerOutletTypeChoices.values())
+        schemas["power_outlet_feedleg_choices"] = json_dumps(PowerOutletFeedLegChoices.values())
+        schemas["interface_type_choices"] = json_dumps(InterfaceTypeChoices.values())
+        schemas["interface_poe_mode_choices"] = json_dumps(InterfacePoEModeChoices.values())
+        schemas["interface_poe_type_choices"] = json_dumps(InterfacePoETypeChoices.values())
+        schemas["port_type_choices"] = json_dumps(PortTypeChoices.values())  # front-ports and rear-ports
```

Comment:
We can simplify this to a mapping of schema section name (e.g. `airflow_choices`) to ChoiceSet class (e.g. `DeviceAirFlowChoices`). We can then use a `for` loop to call `json_dumps()` and generate the template context data. This allows us to separate the rote mapping from the generation logic, which will make it easier to adjust in the future if needed.